src/main/scala/com/fortysevendeg/android/sqlite/metadata/SQLDroidDatabaseMetaData.scala

```diff
@@ -0,0 +1,933 @@
+        case _ => "''"
+      }
+      val ur = ruleMap(rs.getString(updatePos))
+      val dr = ruleMap(rs.getString(deletePos))
+      Option(rs.getString(tablePos)) match {
+        case Some(t) if t.equalsIgnoreCase(table) => Some(exportedKeysSelect(keySeq, table, fkColumn, pkColumn, ur, dr, fkn(table)))
+        case _ => None
+      }
+    }.flatten[String]
+  }
+
+  private[this] def fkn(table: String): Option[String] = {
+    val rsSql = connection.createStatement().executeQuery(sqliteMasterSql(table))
+    if (rsSql.next())
+      PrimaryKey.namedPattern.findFirstMatchIn(rsSql.getString(1)) map { m =>
+        Some(s", '${m.group(1).toLowerCase}' AS fkn")
+      } getOrElse Some(", '' as fkn")
+    else None
+  }
+
+  override def getImportedKeys(
+    catalog: String,
+    schema: String,
+    table: String
+    ): ResultSet = {
+    val statement = connection.createStatement
+    val selectSeq = Try(statement.executeQuery(foreignKeys(table))) match {
+      case Success(resultSet) => generateImportedKeySelect(resultSet, table)
+      case Failure(_) => Seq.empty
+    }
+    Try(statement.close())
+
+    connection.createStatement.executeQuery(
+      importedKeys(
+        catalog = Option(catalog),
+        schema = Option(schema),
+        table = table,
+        importedKeysSelect = if (selectSeq.isEmpty) importedKeysSelectEmpty else selectSeq.mkString(unionAll)))
+  }
+
+  private[this] def generateImportedKeySelect(resultSet: ResultSet, table: String): Seq[String] = {
+    val tablePos = resultSet.findColumn("TABLE")
+    val seqPos = resultSet.findColumn("SEQ")
+    val fkPos = resultSet.findColumn("FROM")
+    val pkPos = resultSet.findColumn("TO")
+    val updatePos = resultSet.findColumn("ON_UPDATE")
+    val deletePos = resultSet.findColumn("ON_DELETE")
+    resultSet.process { rs =>
+      val keySeq = rs.getInt(seqPos) + 1
+      val pkTable = rs.getString(tablePos)
+      val fkColumn = rs.getString(fkPos)
+      val pkColumn = Option(rs.getString(pkPos)) orElse PrimaryKey(connection, table).columns.headOption
+      val updateRule = rs.getString(updatePos)
+      val deleteRule = rs.getString(deletePos)
+      importedKeysSelect(keySeq, pkTable, fkColumn, pkColumn getOrElse "", updateRule, deleteRule)
+    }
+  }
+
+  override def getIndexInfo(
+    catalog: String,
+    schema: String,
+    table: String,
+    unique: Boolean,
+    approximate: Boolean
+    ): ResultSet = {
+    val statement = connection.createStatement
+    Try {
+      statement.executeQuery(pragmaIndexList(table))
+    } match {
+      case Success(resultSet) if !resultSet.isClosed =>
+        val seqPos = resultSet.findColumn("SEQ")
+        val namePos = resultSet.findColumn("NAME")
+        val indexSeq: Seq[(String, Int)] = resultSet.process { rs =>
+          (rs.getString(namePos), rs.getInt(seqPos))
+        }.reverse
+
+        val indexInfoSeq = indexSeq flatMap { tuple =>
+          val (indexName, index) = tuple
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Field name does not match the regular expression '^[a-z][A-Za-z0-9]*$'](https://www.codacy.com/app/47deg/scala-sqlite-droid/file/784801318/issues/source?bid=2442814&fileBranchId=2493741#l271)